codec.go

```diff
@@ -94,6 +96,28 @@ func (c *protoJSONCodec) Unmarshal(binary []byte, message any) error {
 	return options.Unmarshal(binary, protoMessage)
 }
 
+func newProtoJSONCharsetUTF8Codec() Codec {
+	return &protoJSONUTF8Codec{
+		jsonCodec: &protoJSONCodec{},
+	}
+}
+
+type protoJSONUTF8Codec struct {
+	jsonCodec *protoJSONCodec
+}
+
+var _ Codec = (*protoJSONUTF8Codec)(nil)
+
+func (c *protoJSONUTF8Codec) Name() string { return codecNameJSONCharsetUTF8 }
+
+func (c *protoJSONUTF8Codec) Marshal(message any) ([]byte, error) {
+	return c.jsonCodec.Marshal(message)
+}
+
+func (c *protoJSONUTF8Codec) Unmarshal(binary []byte, message any) error {
+	return c.jsonCodec.Unmarshal(binary, message)
+}
```

Comment:
Rather than introducing a new codec, can we have the existing codec keep the name as a field?